packages/app/src/cli/utilities/developer-platform-client/app-management-client.test.ts

```diff
@@ -188,3 +193,200 @@ describe('versionDeepLink', () => {
     expect(got).toEqual('https://dev.shopify.com/dashboard/1/apps/2/versions/3')
   })
 })
+
+describe('deploy', () => {
```

Comment:
Personal opinion.

These tests seem like lengthy restatements of the type signatures, and personally I don't find that they increase my confidence in the code beyond the type signature itself. It's just another thing to change if we need to change the type signature later.

I do see the value in checking the logic in generation of the inputs, since it varies based on inputs to `deploy`. So maybe the tests could be significantly cut down by just focusing on that piece of it, verifying that the inputs are generated correctly. But I think `appManagementRequest` can just be a mock, since the only real concern is that the inputs are right.

Reply:
Hmm, ok, I could simplify the mocked response. There is some post-request code to map the response that it was meant to verify. But it's really an example of trying to do a bit to improve the tests while in there, and not strictly related to this PR.